### PR TITLE
Improve API for BLS crypto functions

### PR DESCRIPTION
**Pull request by dsainati1**
Based on discussion in https://github.com/onflow/flow-go/pull/1605, it seems as though users would like these functions to return a default value in case of a crypto library error. 

______

- [X] Targeted PR against `master` branch
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [X] Updated relevant documentation 
- [X] Re-reviewed `Files changed` in the Github PR explorer
- [X] Added appropriate labels

### CODE REVIEW

File: runtime/cmd/parse/main.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build !wasm
```
Comment (turbolent) [inline runtime/cmd/parse/main.go:1]: Why was this added? Maybe remove it?
```suggestion
```

File: runtime/runtime.go [hunk 2]
```diff
@@ -3277,18 +3287,21 @@ func aggregateBLSPublicKeys(
...
 	var key *PublicKey
 	wrapPanic(func() {
-		key, err = runtimeInterface.AggregateBLSPublicKeys(publicKeys)
+		key, err = runtimeInterface.BLSAggregatePublicKeys(publicKeys)
 	})
 
+	// if the crypto layer produces an error, we have invalid input, return nil
 	if err != nil {
-		return nil, err
+		return interpreter.NilValue{}, nil //nolint
```
Comment (turbolent) [inline runtime/runtime.go:3295]: Maybe only nolint the specific lint that is hit (error is ignored?), instead of all linting

File: runtime/runtime.go [hunk 3]
```diff
@@ -3246,13 +3246,19 @@ func verifyBLSPOP(
...
+
 	var valid bool
 	wrapPanic(func() {
 		valid, err = runtimeInterface.BLSVerifyPOP(publicKey, signature)
 	})
 
+	// if the crypto layer produces an error, we have invalid input, return false
 	if err != nil {
-		return false, err
+		return interpreter.BoolValue(false), nil //nolint
```
Comment (turbolent) [inline runtime/runtime.go:3261]: Same here, maybe don't ignore all linting, just the specific lint

File: runtime/runtime.go [hunk 4]
```diff
@@ -3246,13 +3246,19 @@ func verifyBLSPOP(
 		return false, err
 	}
 
+	// if public key is invalid, verify returns false
+	if !publicKey.IsValid {
+		return interpreter.BoolValue(false), nil
+	}
```
Comment (turbolent) [inline runtime/runtime.go:3252]: The implementation of `BLSVerifyPOP` should instead check `Validated`, and validate the public key if necessary. Then, if the key is invalid, return false. 

```suggestion
```

See a detailed explanation in https://axiomzen.slack.com/archives/CG0B7CJAJ/p1637366567211300?thread_ts=1637289659.195200&cid=CG0B7CJAJ

Comment (tarakby) [reply runtime/runtime.go:3252]: I think the key is guaranteed to be validated already (instantiating a PublicKey in Cadence includes the validation), so it should be safe to just check `publicKey.IsValid`.

**Comment by turbolent** [comment]
@dsainati1 After discussions with @tarakby and @dete in https://axiomzen.slack.com/archives/CG0B7CJAJ/p1637289659195200, I issued up #1268 and @robert-e-davidson3 is going to work on it. Once done, we can rebase this PR and it'll make thinks much simpler (no `IsValid` / `Validated` checking necessary in Cadence or FVM)

**Review by turbolent — APPROVED** [review]
Looks good! Nice
